DatadogSessionReplay/Sources/Writer/Models/EnrichedResource.swift

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import Foundation
+
+/// Extends the resource information with context.
+internal struct EnrichedResource: Codable, Resource {
+    internal struct Context: Codable, Equatable {
+        internal struct Application: Codable, Equatable {
+            let id: String
+        }
+        let type: String
+        let application: Application
+
+        init(_ applicationId: String) {
+            self.type = "resource"
+            self.application = .init(id: applicationId)
+        }
+    }
+    internal var identifier: String
+    internal var data: Data
```

Comment:
How data is represented over the wire?

Reply:
@maxep Let me answer your question here.

It's added to a form this way:
```swift
resources.forEach {
            multipart.addFormData(
                name: "image",
                filename: $0.identifier,
                data: $0.data,
                mimeType: "image/png"
            )
        }
```

both filename and data are default form fields, so it's basically travelling over the wire in the multipart form as an file attachment, with it filename equal to the identifier.

Reply:
This almost seems like a browser API than a native client. I expect image uploading a POST call with body as raw bytes.

Did we consider this option?

Reply:
Sorry, for missing the comment. Sent you the RFC :)